aws/resource_aws_workspaces_directory.go

```diff
@@ -141,17 +141,22 @@ func resourceAwsWorkspacesDirectoryCreate(d *schema.ResourceData, meta interface
 func resourceAwsWorkspacesDirectoryRead(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).workspacesconn
 
-	resp, err := conn.DescribeWorkspaceDirectories(&workspaces.DescribeWorkspaceDirectoriesInput{
-		DirectoryIds: []*string{aws.String(d.Id())},
-	})
+	raw, state, err := workspacesDirectoryRefreshStateFunc(conn, d.Id())()
 	if err != nil {
-		return err
+		return fmt.Errorf("error getting workspaces directory (%s): %s", d.Id(), err)
+	}
+	if state == workspaces.WorkspaceDirectoryStateDeregistered {
+		log.Printf("[WARN] workspaces directory (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
 	}
-	dir := resp.Directories[0]
 
+	dir := raw.(*workspaces.WorkspaceDirectory)
```

Comment:
Don't we return `workspaces.WorkspaceDirectory` right from the refresh function? Why it's raw?